cmd/cfssl/cfssl_test.go

```diff
@@ -1 +1,350 @@
+}
+
+type APISignResponse struct {
+	Response
+	Result struct {
+		CLISignResponse
+	} `json:"result"`
+}
+
+/* Start up the CFSSL local server. */
+func setup() {
+	// Set up a test server using our CA certificate and key.
+	CACert, CAKey, err := testsuite.CreateSelfSignedCert(CARequest)
+	if err != nil {
+		panic(err)
+	}
+	// Set up a test server using our CA certificate and key.
+	serverData := testsuite.CFSSLServerData{CA: CACert, CAKey: CAKey}
+	s, err := testsuite.StartCFSSLServer(addressToTest, portToTest, serverData)
+	server = s
+	if err != nil {
+		setdown()
+		panic(err)
+	}
+}
+
+/* Kill the CFSSL local server.*/
+func setdown() {
+	err := server.Kill()
+	if err != nil {
+		panic(err)
+	}
+	return
+}
+
+/* Return API output*/
+func apiOut(request string, jsonStr []byte, endpoint string, t *testing.T) []byte {
+	setup()
+	req, err := http.NewRequest(request, endpoint, bytes.NewBuffer(jsonStr))
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		setdown()
+		t.Fatalf("Error sending request to client, %v", err)
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		setdown()
+		t.Fatalf("Error reading API response, %v", err)
+	}
+	setdown()
+	return body
+}
+
+/* Return CLI output */
+func cliOut(command *exec.Cmd, t *testing.T) []byte {
+	CLIOutput, err := command.Output()
+	if err != nil {
+		t.Fatalf("Error sending request %s to cli, %v", string(CLIOutput), err)
+	}
+	return []byte(CLIOutput)
+}
+
+/* Format the API/CLI Output into the following structs */
+func apiFormatBundle(body []byte) APIBundleResponse {
+	apiRes := APIBundleResponse{}
+	json.Unmarshal(body, &apiRes)
+	return apiRes
+}
+
+func apiFormatScan(body []byte) APIScanResponse {
+	apiRes := APIScanResponse{}
+	json.Unmarshal(body, &apiRes)
+	return apiRes
+}
+
+func apiFormatCertInfo(body []byte) APICertInfoResponse {
+	apiRes := APICertInfoResponse{}
+	json.Unmarshal(body, &apiRes)
+	return apiRes
+}
+
+func apiFormatSign(body []byte) APISignResponse {
+	apiRes := APISignResponse{}
+	json.Unmarshal(body, &apiRes)
+	return apiRes
+}
+
+func cliFormatBundle(body []byte) CLIBundleResponse {
+	cliRes := CLIBundleResponse{}
+	json.Unmarshal(body, &cliRes)
+	return cliRes
+}
+
+func cliFormatScan(body []byte) CLIScanResponse {
+	cliRes := CLIScanResponse{}
+	index := strings.Index(string(body), "{")
+	cliout := string(body)[index:]
+	json.Unmarshal([]byte(cliout), &cliRes)
+	return cliRes
+}
+
+func cliFormatCertInfo(body []byte) CLICertInfoResponse {
+	cliRes := CLICertInfoResponse{}
+	json.Unmarshal(body, &cliRes)
+	return cliRes
+}
+
+func responseUnmarshal(body []byte) Response {
+	res := Response{}
+	json.Unmarshal(body, &res)
+	return res
+}
+
+/* START OF TESTS*/
+
+/* Testing the Bundle API against CFSSL Bundle CLI */
+func TestBundle(t *testing.T) {
+	var jsonStr = []byte(`{"domain": "example.com"}`)
+	bundleAPIResp := apiFormatBundle(apiOut("POST", jsonStr, postStr+"/api/v1/cfssl/bundle", t))
```

Comment:
Since it seems like every all to `apiOut` uses `postStr + "/api/v1/cfssl"`, perhaps `apiOut` could put those in automatically to make the code a bit cleaner